packages/mgt/src/graph/graph.photos.ts

```diff
@@ -121,14 +120,36 @@ export async function getContactPhoto(graph: IGraph, contactId: string): Promise
 export async function getUserPhoto(graph: IGraph, userId: string): Promise<string> {
   let cache: CacheStore<CachePhoto>;
   let photoDetails: CachePhoto;
+
   if (photosCacheEnabled()) {
     cache = CacheService.getCache<CachePhoto>(cacheSchema, userStore);
     photoDetails = await cache.getValue(userId);
     if (photoDetails && getPhotoInvalidationTime() > Date.now() - photoDetails.timeCached) {
       return photoDetails.photo;
+    } else if (photoDetails) {
+      // there is a photo in the cache, but it's stale
+      graph
+        .api(`users/${userId}/photo`)
+        .get()
+        .then(
+          async response => {
+            if (response && response['@odata.mediaEtag'] !== photoDetails.eTag) {
+              photoDetails = await getPhotoForResource(graph, `users/${userId}`, ['user.readbasic.all']);
+            }
+            // put current or new image into the cache to update the timestamp
+            cache.putValue(userId, photoDetails);
+            return photoDetails ? photoDetails.photo : null;
+          },
+          error => {
+            cache.putValue(userId, {});
+            return null;
+          }
+        );
```

Comment:
You should use await and try/catch here if possible, makes the code look cleaner. Also, see if you can find a way to call `getPhotoForResource` only below and not repeat the same code

Reply:
Can you explain what I should try/catch here? Not sure what errors need to be accounted for.